### DISCUSSION TITLE
Nginx unit's analog for fpm max_children

### DISCUSSION THREAD

**Discussion by devel96** [nginx/unit#1029] [Support]
Hello. Could you say what can I use instead of these parameters in nginx-unit ? 
max_children
start_servers
min_spare_servers
max_spare_servers

And what to do if my application's rps is like 10-15000 ?

**Comment by ac000** [comment]
This is perhaps better served as a discussion...

**Comment by ac000** [comment]
See [here](https://unit.nginx.org/configuration/#configuration-proc-mgmt-prcs) for an overview of Units application process settings.

I don't think there are completely analogous settings in Unit, Units architecture is a little different to nginx's.

**Reply by devel96** [reply]
Thanks for answer. But to be honest load tests with nginx unit are not effective (. But they work fine with fpm + nginx. Is it ok or maybe I should  consider other settings for unit ?

**Reply by tippexs** [reply]
Hi @devel96 sorry for the late response! Really sorry! 
Can you please share your Unit configuration. 

The link @ac000 just shared is a good first touch point if we are talking about what figures you should use you can start with something like

```json
"applications": {
    "php": {
      "type": "php",
      "limits": {
        "requests": 1000,
        "timeout": 60
      },
      "processes": {
        "max": 50,
        "spare": 15,
        "idle_timeout": 30
      },
      "user": "www-data",
      "group": "www-data",
      "working_directory": "/var/www/",
      "root": "/var/www/public",
      "script": "index.php",
      "index": "index.php"
    }
  }
```
The actual values are never a one size fits all approach. It depends on how many applications are running on this single Unit instance and what are the overall resources look like.
